Userland/Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -2344,6 +2344,50 @@ void Element::scroll_by(HTML::ScrollToOptions options)
     scroll(options);
 }
 
+// https://drafts.csswg.org/cssom-view-1/#dom-element-checkvisibility
+bool Element::check_visibility(Optional<CheckVisibilityOptions> options)
+{
+    // 1. If this does not have an associated box, return false.
+    if (!paintable_box())
```

Comment:
I think you need to call `document().update_layout()` before doing this, otherwise this isn't guaranteed to be up to date.

Reply:
Ah, I see thats being done in other places too, I've updated it!